extensions/amp-viralize-player/0.1/amp-viralize-player.js

```diff
@@ -0,0 +1,378 @@
+    script.async = true;
+    this.container_.appendChild(script);
+
+    return new Promise((resolve) => {
+      this.win.vpt.queue.push(() => {
+        this.win.vpt.on(
+          {event: this.win.vpt.EVENTS.PLAYER_READY, zid: this.zid_},
+          () => {
+            this.playerReady_ = true;
+            this.iframe_ = Array.from(
+              this.element.ownerDocument.getElementsByTagName('iframe')
+            ).find((el) => el.id.match(/vr-[A-Za-z0-9]*-player-iframe/));
+            this.dispatchEvent_(VideoEvents.LOAD);
+            this.executeCmdQueue_();
+            resolve();
+          }
+        );
+
+        this.win.vpt.on(
+          {event: this.win.vpt.EVENTS.START, zid: this.zid_},
+          () => {
+            this.dispatchEvent_(VideoEvents.AD_START);
+          }
+        );
+
+        this.win.vpt.on(
+          {event: this.win.vpt.EVENTS.COMPLETE, zid: this.zid_},
+          () => {
+            this.dispatchEvent_(VideoEvents.AD_END);
+          }
+        );
+
+        this.win.vpt.on(
+          {event: this.win.vpt.EVENTS.SKIP, zid: this.zid_},
+          () => {
+            this.dispatchEvent_(VideoEvents.AD_END);
+          }
+        );
+
+        this.win.vpt.on(
+          {event: this.win.vpt.EVENTS.CONTENT_END, zid: this.zid_},
+          () => {
+            this.dispatchEvent_(VideoEvents.ENDED);
+          }
+        );
+
+        this.win.vpt.on(
+          {event: this.win.vpt.EVENTS.PLAYING, zid: this.zid_},
+          () => {
+            this.dispatchEvent_(VideoEvents.PLAYING);
+          }
+        );
+
+        this.win.vpt.on(
+          {event: this.win.vpt.EVENTS.PAUSED, zid: this.zid_},
+          () => {
+            this.dispatchEvent_(VideoEvents.PAUSE);
+          }
+        );
+
+        this.win.vpt.on(
+          {event: this.win.vpt.EVENTS.MUTED, zid: this.zid_},
+          () => {
+            this.dispatchEvent_(VideoEvents.MUTED);
+          }
+        );
+
+        this.win.vpt.on(
+          {event: this.win.vpt.EVENTS.UNMUTED, zid: this.zid_},
+          () => {
+            this.dispatchEvent_(VideoEvents.UNMUTED);
+          }
+        );
+      });
+    });
+  }
+
+  /** @override */
+  unlayoutCallback() {
+    if (this.container_) {
+      removeElement(this.container_);
+      this.container_ = null;
+    }
+    return true;
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout === Layout.RESPONSIVE;
+  }
+
+  /** @override */
+  viewportCallback(visible) {
+    this.dispatchEvent_(VideoEvents.VISIBILITY, {visible});
+  }
+
+  /** @override */
+  pauseCallback() {
+    this.pause();
+  }
+
+  /**
+   * VideoInterface implementations
+   */
+
+  /** @override */
+  supportsPlatform() {
+    return true;
+  }
+
+  /** @override */
+  isInteractive() {
+    return true;
+  }
+
+  /** @override */
+  getCurrentTime() {
+    // Not supported.
+    return 0;
+  }
+
+  /** @override */
+  getDuration() {
+    // Not supported.
+    return 1;
```

Comment:
Are we sure is correct to return 1 if `duration` is not supported?

Reply:
This has been copy pasted from another player. I check what other players which do not support duration do, and I found that someone return 0, someone 1, someone NaN. I think that it doesn't matter because documentation says:
```
  Total duration of the video in seconds
  This is used for analytics metadata.
```
So I think that this value is used if a client use AMP analytics API, and in that case, any value that we return should not be relied.